### PR TITLE
[AIRFLOW-5139] Allow custom ES configs

### PR DESCRIPTION
**Pull request by dimberman**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5139
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

While attempting to create a self-signed TLS connection between airflow
and ES, we discovered that airflow does now allow users to modify the
SSL state of the elasticsearchtaskhandler. This commit will allow users
to define ES settings in the airflow.cfg

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Comment by dimberman** [comment]
Huge shoutout to @potiuk this new testing environment has made writing fixes 10x easier :)

**Comment by dimberman** [comment]
@potiuk Somethings seems to be wrong with the check-licenses. It works fine locally, but here it's complaining about a corrupted /tmp/rats.jar. Have you seen this elsewhere/do you think this might be tied to an unversioned jar?

**Comment by potiuk** [comment]
@dimberman  Yeah. It seems that there was a corrupted jar which got part of the Docker image. Seems that the maven storage for the RAT repo got really unstable (happened to me several times today). There is a fix being built right now https://travis-ci.org/apache/airflow/builds/569350104  from this PR https://github.com/apache/airflow/pull/5759. 

It's a permanent fix - it will both - prevent it in the future (Md5 check + testing the .jar file) but also once the Docker image is built - it will not be broken again in next build.

**Comment by potiuk** [comment]
Hey @dimberman -> it should all be fixed. The image is rebuilt in Dockerhub and all the subsequent builds will use the "proper" rat from the Docker image :)

**Review by feng-tao — APPROVED** [review]

File: airflow/utils/log/es_task_handler.py [hunk 1]
```diff
@@ -54,7 +55,8 @@ class ElasticsearchTaskHandler(FileTaskHandler, LoggingMixin):
     def __init__(self, base_log_folder, filename_template,
                  log_id_template, end_of_log_mark,
                  write_stdout, json_format, json_fields,
-                 host='localhost:9200'):
+                 host='localhost:9200',
+                 es_kwargs=conf.getsection("elasticsearch_configs") or {}):
```
Comment (kaxil) [inline airflow/utils/log/es_task_handler.py:59]: Can we add this to https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg as well, please?

**Review by kaxil — CHANGES_REQUESTED** [review]
Small comment regarding adding it to `airflow.cfg`. Other than that LGTM

File: airflow/config_templates/default_airflow.cfg [hunk 2]
```diff
@@ -591,6 +591,11 @@ json_format = False
 # Log fields to also attach to the json output, if enabled
 json_fields = asctime, filename, lineno, levelname, message
 
+[elasticsearch_configs]
+
+use_ssl = False
+verify_certs = False
```
Comment (schnie) [inline airflow/config_templates/default_airflow.cfg]: @dimberman wondering if we should mirror the defaults that the es client uses here so we don't change any behavior. `verify_certs` is `True` by default here: https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/connection/http_requests.py#L47-L48

Comment (dimberman) [reply airflow/config_templates/default_airflow.cfg]: @schnie good catch, thought they both defaulted to false

File: airflow/config_templates/default_airflow.cfg [hunk 3]
```diff
@@ -591,6 +591,11 @@ json_format = False
 # Log fields to also attach to the json output, if enabled
 json_fields = asctime, filename, lineno, levelname, message
 
+[elasticsearch_configs]
+
+use_ssl = False
+verify_certs = True
+
```
Comment (kaxil) [inline airflow/config_templates/default_airflow.cfg:598]: **ca_certs** is missing ??

Comment (dimberman) [reply airflow/config_templates/default_airflow.cfg:598]: These are just example configs. Users can add anything in the elasticsearch SDK

**Review by kaxil — APPROVED** [review]

**Pull request merged**